front/src/modules/simulationResult/helpers/utils.ts

```diff
@@ -0,0 +1,113 @@
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainSchedule/utils';
+
+export const filterSignalUpdatesByPosition = (
+  signals: ProjectPathTrainResult['signal_updates'],
+  cutPositionStart: number,
+  cutPositionEnd: number
+): ProjectPathTrainResult['signal_updates'] =>
+  signals
+    .map((signal) => {
+      if (signal.position_start < cutPositionStart && signal.position_end > cutPositionStart) {
+        const interpolationFactor =
+          (cutPositionStart - signal.position_start) /
+          (signal.position_end - signal.position_start);
+        return {
+          ...signal,
+          position_start: cutPositionStart,
+          time_start:
+            signal.time_start + (signal.time_end - signal.time_start) * interpolationFactor,
+        };
+      }
+      if (signal.position_start < cutPositionEnd && signal.position_end > cutPositionEnd) {
+        const interpolationFactor =
+          (signal.position_end - cutPositionEnd) / (signal.position_end - signal.position_start);
+        return {
+          ...signal,
+          position_end: cutPositionEnd,
+          time_end: signal.time_end - (signal.time_end - signal.time_start) * interpolationFactor,
+        };
+      }
+      return signal;
+    })
+    .filter(
+      (signal) => signal.position_start >= cutPositionStart && signal.position_end <= cutPositionEnd
+    );
+
+export const filterSpaceTimeCurvesByPosition = (
+  spaceTimeCurves: ProjectPathTrainResult['space_time_curves'],
+  cutPositionStart: number,
+  cutPositionEnd: number
+): ProjectPathTrainResult['space_time_curves'] =>
+  spaceTimeCurves.map((spaceTimeCurve) => {
+    let { positions, times } = { ...spaceTimeCurve };
+
+    // If the position where we want to cut the curve isn't already in the curve positions,
+    // We need to interpolate the corresponding time and it and the position the curve
+
+    if (positions.at(0)! <= cutPositionStart && !positions.includes(cutPositionStart)) {
+      const interpolatedTime = interpolateValue(
+        { positions, speeds: [], times },
+        cutPositionStart,
+        'times'
+      );
+
+      positions = [cutPositionStart, ...positions];
+      times = [interpolatedTime, ...times];
+    }
+
+    if (positions.at(-1)! >= cutPositionEnd && !positions.includes(cutPositionEnd)) {
+      const interpolatedTime = interpolateValue(
+        { positions, speeds: [], times },
+        cutPositionEnd,
+        'times'
+      );
+
+      positions = [...positions, cutPositionEnd];
+      times = [...times, interpolatedTime];
+    }
+
+    const filteredIndex = positions
+      .map((position, index) =>
+        position >= cutPositionStart && position <= cutPositionEnd ? index : -1
+      )
+      .filter((index) => index !== -1);
+
+    return {
+      positions: filteredIndex.map((index) => positions[index]),
+      times: filteredIndex.map((index) => times[index]),
+    };
+  });
+
+export const filterConflictsByPosition = (
```

Comment:
What do you think of this version ? In this function, we only go through the array once 
```
export const cutConflictsByPosition = (
  conflicts: Conflict[],
  cutPositionStart: number,
  cutPositionEnd: number
) => {
  const results: Conflict[] = [];

  for (const conflict of conflicts) {
    const { spaceStart, spaceEnd, timeStart, timeEnd } = conflict;
    if (spaceStart < cutPositionStart && spaceEnd > cutPositionEnd) {
      continue;
    }
    if (spaceStart < cutPositionStart) {
      const interpolationFactor = (cutPositionStart - spaceStart) / (spaceEnd - spaceStart);
      results.push({
        spaceStart: cutPositionStart,
        spaceEnd,
        timeStart: timeStart + (timeEnd - timeStart) * interpolationFactor,
        timeEnd,
      });
      continue;
    }
    if (spaceEnd > cutPositionEnd) {
      const interpolationFactor = (spaceEnd - cutPositionEnd) / (spaceEnd - spaceStart);
      results.push({
        spaceStart,
        spaceEnd: cutPositionEnd,
        timeStart,
        timeEnd: timeEnd - (timeEnd - timeStart) * interpolationFactor,
      });
      continue;
    }
    results.push(conflict);
  }

  return results;
};
```